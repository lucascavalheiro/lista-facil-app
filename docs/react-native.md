# Getting Started with React Native

## Setting Up Your Machine for App Development

- iOS (You require a Mac to develop iOS Apps) Simply download XCode
- [Android on Mac](https://medium.com/pvtl/react-native-android-development-on-mac-ef7481f65e47#.w2hel88zy)
- Android on Windows - coming soon...

## Install React Native

[React Native - Get Started Guide](https://facebook.github.io/react-native/docs/getting-started.html) 

## Learn React

[React Native Express](http://www.reactnativeexpress.com/) is a great site to get you started, specifically:

- [Get your head around ES6](http://www.reactnativeexpress.com/es6)
- [What is JSX?](http://www.reactnativeexpress.com/jsx)
- [What are Components?](http://www.reactnativeexpress.com/components)
- [React State](http://www.reactnativeexpress.com/data_component_state)
- [Redux](http://www.reactnativeexpress.com/redux)

Once you've got your head around the basics, checkout the [React Native](https://facebook.github.io/react-native/) website, specifically:

- Go through ['The Basics'](https://facebook.github.io/react-native/docs/props.html)
- Gain an understanding of the [components](https://facebook.github.io/react-native/docs/activityindicator.html) React Native provides out of the box
