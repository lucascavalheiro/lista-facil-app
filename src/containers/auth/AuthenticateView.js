/**
 * Authenticate Screen
 *  - Entry screen for all authentication
 *  - User can tap to login, forget password, signup...
 *
 * React Native Starter App
 * https://github.com/mcnamee/react-native-starter-app
 */
import React, { Component } from 'react';
import {
  View,
  Image,
  StyleSheet,
} from 'react-native';
import { Actions } from 'react-native-router-flux';

// Consts and Libs
import { AppStyles, AppSizes, AppColors } from '@theme/';

// Components
import { Spacer, Text, Button } from '@ui/';

/* Styles ==================================================================== */
const styles = StyleSheet.create({
  background: {
    backgroundColor: AppColors.brand.primary,
    height: AppSizes.screen.height,
    width: AppSizes.screen.width,
  },
  logo: {
    width: AppSizes.screen.width * 0.85,
    resizeMode: 'contain',
  },
  whiteText: {
    color: '#FFF',
  },
});

/* Component ==================================================================== */
class Authenticate extends Component {
  static componentName = 'Authenticate';

  render = () => (
    <View style={[AppStyles.containerCentered, AppStyles.container, styles.background]}>
      <Image
        source={require('../../images/logo.png')}
        style={[styles.logo]}
      />

      <View style={[AppStyles.row, AppStyles.paddingHorizontal]}>
        <View style={[AppStyles.flex1]}>
          <Button
            title={'Login'}
            icon={{ name: 'lock' }}
            onPress={Actions.login}
            backgroundColor={'#CB009E'}
          />
        </View>
      </View>

      <Spacer size={10} />

      <View style={[AppStyles.row, AppStyles.paddingHorizontal]}>
        <View style={[AppStyles.flex1]}>
          <Button
            title={'Sign up'}
            icon={{ name: 'face' }}
            onPress={Actions.signUp}
            backgroundColor={'#CB009E'}
          />
        </View>
      </View>

      <Spacer size={15} />

      <Text p style={[AppStyles.textCenterAligned, styles.whiteText]}>
        - or -
      </Text>

      <Spacer size={10} />

      <View style={[AppStyles.row, AppStyles.paddingHorizontal]}>
        <View style={[AppStyles.flex1]} />
        <View style={[AppStyles.flex2]}>
          <Button
            small
            title={'Skip'}
            onPress={Actions.app}
            raised={false}
            backgroundColor={'rgba(255,255,255,0.2)'}
          />
        </View>
        <View style={[AppStyles.flex1]} />
      </View>

      <Spacer size={40} />
    </View>
  )
}

/* Export Component ==================================================================== */
export default Authenticate;
