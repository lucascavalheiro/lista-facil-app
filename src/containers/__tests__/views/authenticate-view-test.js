/**
 * Test to check if the component renders correctly
 */
/* global it expect */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import AuthenticateView from '@containers/auth/AuthenticateView';

it('AuthenticateView renders correctly', () => {
  const tree = renderer.create(
    <AuthenticateView />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
