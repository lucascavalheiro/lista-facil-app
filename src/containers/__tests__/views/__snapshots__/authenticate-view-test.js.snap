// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthenticateView renders correctly 1`] = `
<Image
  source={1}
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "#E9EBEE",
        "flex": 1,
        "flexDirection": "column",
        "position": "relative",
      },
      Object {
        "backgroundColor": "transparent",
        "height": 1334,
        "width": 750,
      },
    ]
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "resizeMode": "contain",
          "width": 637.5,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "left": 0,
          "right": 0,
        },
        Object {
          "paddingHorizontal": 20,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "marginLeft": 15,
                "marginRight": 15,
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "#0E4EF8",
              },
              Object {
                "borderRadius": 2,
              },
              Object {
                "shadowColor": "rgba(0,0,0, .4)",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "shadowOpacity": 1,
                "shadowRadius": 1,
              },
              Object {
                "padding": 12,
              },
              Object {
                "marginLeft": 0,
                "marginRight": 0,
                "padding": 12,
              },
              undefined,
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 14,
                },
                Array [
                  Object {
                    "marginRight": 10,
                  },
                  undefined,
                ],
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": undefined,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#fff",
                  },
                  Object {
                    "fontSize": 17.5,
                  },
                  Object {
                    "fontSize": 14,
                  },
                  undefined,
                  Object {
                    "fontWeight": "bold",
                  },
                  Object {
                    "fontFamily": "HelveticaNeue",
                  },
                ],
              ]
            }
          >
            Login
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "height": 1,
        "left": 0,
        "marginTop": 9,
        "right": 0,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "left": 0,
          "right": 0,
        },
        Object {
          "paddingHorizontal": 20,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "marginLeft": 15,
                "marginRight": 15,
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "#0E4EF8",
              },
              Object {
                "borderRadius": 2,
              },
              Object {
                "shadowColor": "rgba(0,0,0, .4)",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "shadowOpacity": 1,
                "shadowRadius": 1,
              },
              Object {
                "padding": 12,
              },
              Object {
                "marginLeft": 0,
                "marginRight": 0,
                "padding": 12,
              },
              undefined,
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 14,
                },
                Array [
                  Object {
                    "marginRight": 10,
                  },
                  undefined,
                ],
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": undefined,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#fff",
                  },
                  Object {
                    "fontSize": 17.5,
                  },
                  Object {
                    "fontSize": 14,
                  },
                  undefined,
                  Object {
                    "fontWeight": "bold",
                  },
                  Object {
                    "fontFamily": "HelveticaNeue",
                  },
                ],
              ]
            }
          >
            Sign up
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "height": 1,
        "left": 0,
        "marginTop": 14,
        "right": 0,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    h1={false}
    h2={false}
    h3={false}
    h4={false}
    h5={false}
    onPress={null}
    p={true}
    style={
      Array [
        Object {
          "color": "#222222",
          "fontFamily": "HelveticaNeue",
          "fontSize": 14,
          "fontWeight": "500",
          "lineHeight": 18,
          "marginBottom": 8,
        },
        Array [
          Object {
            "textAlign": "center",
          },
          Object {
            "color": "#FFF",
          },
        ],
      ]
    }
  >
    - or -
  </Text>
  <View
    style={
      Object {
        "height": 1,
        "left": 0,
        "marginTop": 9,
        "right": 0,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "left": 0,
          "right": 0,
        },
        Object {
          "paddingHorizontal": 20,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "flex": 2,
          },
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "marginLeft": 15,
                "marginRight": 15,
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "#CB009E",
              },
              Object {
                "borderRadius": 2,
              },
              false,
              Object {
                "padding": 12,
              },
              Object {
                "marginLeft": 0,
                "marginRight": 0,
                "padding": 8,
              },
              undefined,
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": undefined,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#fff",
                  },
                  Object {
                    "fontSize": 17.5,
                  },
                  Object {
                    "fontSize": 12,
                  },
                  undefined,
                  Object {
                    "fontWeight": "bold",
                  },
                  Object {
                    "fontFamily": "HelveticaNeue",
                  },
                ],
              ]
            }
          >
            Skip
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "height": 1,
        "left": 0,
        "marginTop": 39,
        "right": 0,
      }
    }
  />
</Image>
`;
